VoodooTrackpoint/VoodooTrackpoint.cpp

```diff
@@ -18,9 +18,22 @@ bool VoodooTrackpoint::start(IOService *provider) {
         return false;
     }
     
+    setProperty(VOODOO_TRACKPOINT_IDENTIFIER, kOSBooleanTrue);
+    
+    if (!provider->open(this)) {
```

Comment:
I wonder why we have to do this here. Why do we have to call into the provider here?

Reply:
Otherwise the provider gets no notification that it attached. You could do `getClient` but there is no guarantee that it would be VoodooTrackpoint. By opening and closing, we know for sure that it is there or not and have a definitive pointer to it.